Scenarios/run_Scenario.bat

```diff
@@ -1,74 +1,211 @@
-   set Major=%%a
-   set Minor=%%b
-   set Revision=%%c
+FOR /F "tokens=1,2,3 delims=." %%a in ("!ReleaseVersion!") DO (
+    SET Major=%%a
+    SET Minor=%%b
+    SET Revision=%%c
 )
 
-echo esmini version is: !Major!.!Minor!.!Revision!
+ECHO esmini version is: !Major!.!Minor!.!Revision!
 
-set supported=1
+SET supported=1
 
-if !Major! LSS 1 set supported=0
-if !Major! EQU 1 (
-    if !Minor! LSS 7 set supported=0
-    if !Minor! EQU 7 (
-        if !Revision! LSS 13 set supported=0
+IF !Major! LSS 1 SET supported=0
+IF !Major! EQU 1 (
+    IF !Minor! LSS 7 SET supported=0
+    IF !Minor! EQU 7 (
+        IF !Revision! LSS 13 SET supported=0
         )
     )
 )
 
-if !supported! EQU 0 (
-    echo Unsupported esmini version.
-    goto finish
+IF !supported! EQU 0 (
+    ECHO Unsupported esmini version.
+    GOTO finish
 )
 
-:: Section 2: Parse OpenScenario files in same folder.
-set x=0
-for /f "delims=" %%a in ('dir /s /b *.xosc') do (
-   set /A "x+=1"
-   set list[!x!]=%%a
+SET x=0
+FOR /f "delims=" %%a in ('dir /s /b *.xosc') DO (
+    SET /A "x+=1"
+    SET list[!x!]=%%a
 )
-set Filesx=%x%
+SET Filesx=%x%
 
-:again
-ECHO ====================================
-ECHO List of OpenSCENARIO (XOSC) files.
-ECHO ====================================
-rem Display array elements
-for /L %%i in (1,1,%Filesx%) do echo [%%i]: "!list[%%i]!"
-ECHO ====================================
-ECHO Select and play a scenario by number.
+ECHO ####################################
+ECHO #       ESMINI: SIMULATION         #
+ECHO ####################################
+
+ECHO Supported OpenSCENARIO (XOSC) files:
+FOR /L %%i in (1,1,%Filesx%) DO ECHO [%%i]: "!list[%%i]!"
+:again_esmini
+ECHO ------------------------------------
+ECHO Select a scenario by number.
 ECHO Abort with 0.
-ECHO ====================================
-set id=0
-set /p id="Enter ID: "
+SET id=0
+SET /p id="Enter ID: "
 
-if %id% GEQ 1 (
-    if %id% LEQ !x! (
+IF %id% GEQ 1 (
+    IF %id% LEQ !x! (
         "%ESMINI%\EgoSimulator.exe" --window 100 200 1024 576  --osc !list[%id%]! --trails off
-        GOTO again
+        GOTO again_esmini
+    )
+)
+
+GOTO finish
+
+:OpenPASS
+
+SET num_supported_scenario_version=7
+SET list_supported_scenario_version[1]=4.1_1
+SET list_supported_scenario_version[2]=4.2_1
+SET list_supported_scenario_version[3]=4.2_2
+SET list_supported_scenario_version[4]=4.2_4
+SET list_supported_scenario_version[5]=4.5_1
+SET list_supported_scenario_version[6]=4.5_2
+SET list_supported_scenario_version[7]=4.6_1
+
+ECHO ####################################
+ECHO #    OPENPASS: SYSTEM SETTINGS     #
+ECHO ####################################
+
+SET dir_path=%~dp0
+
+IF DEFINED OPENPASS (
+    ECHO OPENPASS path found.
+    ECHO OPENPASS path: %OPENPASS%
+) ELSE (
+    ECHO OPENPASS path not found. 
+    ECHO Please create OPENPASS environment variable, which directs to the installation directory of openPASS and restart shell.
```

Comment:
and restart script

Reply:
Resolved in a2aaf59093454ee9a5886ba5cfb856c760cdf850